lib/msal-browser/src/interaction_handler/RedirectHandler.ts

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { IInteractionHandler } from "./IInteractionHandler";
+import { BrowserAuthError } from "../error/BrowserAuthError";
+import { StringUtils, AuthorizationCodeModule, TemporaryCacheKeys, AuthenticationParameters, AuthError, AuthResponse, TokenResponse } from "msal-common";
+import { AuthCallback } from "../app/PublicClientApplication";
+import { BrowserConfigurationAuthError } from "../error/BrowserConfigurationAuthError";
+import { BrowserStorage } from "../cache/BrowserStorage";
+
+export class RedirectHandler extends IInteractionHandler {
+
+    private authCallback: AuthCallback;
+
+    constructor(authCodeModule: AuthorizationCodeModule, storageImpl: BrowserStorage,  redirectCallback: AuthCallback) {
+        super(authCodeModule, storageImpl);
+        if (!redirectCallback) {
+            throw BrowserConfigurationAuthError.createRedirectCallbacksNotSetError();
+        }
+        this.authCallback = redirectCallback;
+    }
+
+    /**
+     * Redirects window to given URL.
+     * @param urlNavigate 
+     */
+    showUI(authRequest: AuthenticationParameters): void {
+        this.browserStorage.setItem(TemporaryCacheKeys.ORIGIN_URI, window.location.href);
+        this.authModule.createLoginUrl(authRequest).then((urlNavigate) => {
+            // Navigate if valid URL
+            if (urlNavigate && !StringUtils.isEmpty(urlNavigate)) {
+                this.authModule.logger.infoPii("Navigate to:" + urlNavigate);
+                window.location.assign(urlNavigate);
+            }
+            else {
+                this.authModule.logger.info("Navigate url is empty");
+                throw BrowserAuthError.createEmptyRedirectUriError();
+            }
+        });
+    }
+
+    /**
+     * Handle authorization code response in the window.
+     * @param hash 
+     */
+    handleCodeResponse(hash: string, navigateToLoginRequestUrl?: boolean): void {
+        // retrieve the hash
+        const locationHash = hash || window.location.hash;
```

Comment:
Will this ever be null? Seems nonsensical.

Reply:
removed the || assignment, added null check for hash